src/main/kotlin/org/jetbrains/research/testspark/helpers/psi/kotlin/KotlinPsiClassWrapper.kt

```diff
@@ -0,0 +1,130 @@
+import org.jetbrains.research.testspark.helpers.psi.PsiMethodWrapper
+
+class KotlinPsiClassWrapper(private val psiClass: KtClassOrObject) : PsiClassWrapper {
+    override val name: String get() = psiClass.name ?: ""
+
+    override val qualifiedName: String get() = psiClass.fqName!!.asString() ?: ""
+
+    override val text: String? get() = psiClass.text
+
+    override val methods: List<PsiMethodWrapper>
+        get() = psiClass.body?.functions?.filter { it.name != null }?.map { KotlinPsiMethodWrapper(it) } ?: emptyList()
+
+    override val allMethods: List<PsiMethodWrapper> get() = methods
+
+    override val superClass: PsiClassWrapper?
+        get() {
+            // Get the superTypeListEntries of the Kotlin class
+            val superTypeListEntries = psiClass.superTypeListEntries
+            // Find the superclass entry (if any)
+            val superClassEntry = superTypeListEntries.firstOrNull()
+            // Resolve the superclass type reference to a PsiClass
+            val superClassTypeReference = superClassEntry?.typeReference
+            val superClassDescriptor = superClassTypeReference?.let {
+                val bindingContext = it.analyze()
+                bindingContext[BindingContext.TYPE, it]
+            }
+            val superClassPsiClass = superClassDescriptor?.constructor?.declarationDescriptor?.let { descriptor ->
+                DescriptorToSourceUtils.getSourceFromDescriptor(descriptor) as? KtClass
+            }
+            return if (psiClass.fqName != null) {
+                superClassPsiClass?.let { KotlinPsiClassWrapper(it) }
+            } else {
+                null
+            }
+        }
+
+    override val virtualFile: VirtualFile get() = psiClass.containingFile.virtualFile
+
+    override val containingFile: PsiFile get() = psiClass.containingFile
+
+    override val fullText: String
+        get() {
+            var fullText = ""
+            val fileText = psiClass.containingFile.text
+
+            // get package
+            packagePattern.findAll(fileText, 0).map {
+                it.groupValues[0]
+            }.forEach {
+                fullText += "$it\n\n"
+            }
+
+            // get imports
+            importPattern.findAll(fileText, 0).map {
+                it.groupValues[0]
+            }.forEach {
+                fullText += "$it\n"
+            }
+
+            // Add class code
+            fullText += psiClass.text
+
+            return fullText
+        }
+
+    override val classType: ClassType
+        get() {
+            return when {
+                psiClass is KtObjectDeclaration -> ClassType.OBJECT
+                psiClass.isInterfaceClass() -> ClassType.INTERFACE
+                psiClass.hasModifier(KtTokens.ABSTRACT_KEYWORD) -> ClassType.ABSTRACT_CLASS
+                psiClass.isData() -> ClassType.DATA_CLASS
+                psiClass.annotationEntries.any { it.text == "@JvmInline" } -> ClassType.INLINE_VALUE_CLASS
```

Comment:
According to the [docs](https://kotlinlang.org/docs/inline-classes.html), the inline value class may not have a `@JvmInline` annotation. 

Will this `when` construct classify the following example as `INLINE_VALUE_CLASS`?
```kotlin
value class Password(private val s: String)
```

Reply:
This is an inlina_value_class
```
@JvmInline
value class MyInlineClass(val value: Int)
```

And your example will not compile 
<img width="842" alt="image" src="https://github.com/JetBrains-Research/TestSpark/assets/70476032/060ea777-5924-4b3f-8540-cd60a01fe309">